src/metabase/db/liquibase.clj

```diff
@@ -197,31 +203,39 @@
   "Check and make sure the database isn't locked. If it is, sleep for 2 seconds and then retry several times. There's a
   chance the lock will end up clearing up so we can run migrations normally."
   [^Liquibase liquibase]
-  (u/auto-retry 5
-    (when (migration-lock-exists? liquibase)
-      (Thread/sleep 2000)
-      (throw
-       (LockException.
-        (str
-         (trs "Database has migration lock; cannot run migrations.")
-         " "
-         (trs "You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.")))))))
+  (if (migration-lock-exists? liquibase)
+    (do
+     (log/info "Database has migration lock. Waiting for lock to be cleared...")
+     (Thread/sleep 2000)
+     (u/auto-retry 4
+       (when (migration-lock-exists? liquibase)
+         (Thread/sleep 2000)
+         (throw
+          (LockException.
+           (str
+            (trs "Database has migration lock; cannot run migrations.")
+            " "
+            (trs "You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.")))))))
+    (log/info "No migration lock found.")))
 
 (defn migrate-up-if-needed!
   "Run any unrun `liquibase` migrations, if needed."
-  [^Liquibase liquibase]
+  [^Liquibase liquibase ^DataSource data-source]
   (log/info (trs "Checking if Database has unrun migrations..."))
-  (if (seq (unrun-migrations liquibase))
+  (if (seq (unrun-migrations data-source))
     (do
-     (log/info (trs "Database has unrun migrations. Waiting for migration lock to be cleared..."))
+     (log/info (trs "Database has unrun migrations. Checking if migraton lock is cleared..."))
```

Comment:
Out of scope nit on language - taking about a lock being created or cleared is a quirk in my opinion, speaking to the liquibase internals. I'd rather say things like "checking if the lock is taken".

Reply:
makes sense! 4cf508c09a